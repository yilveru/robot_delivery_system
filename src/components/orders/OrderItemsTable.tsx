"use client";

import React from "react";

import {
  Table,
  TableBody,
  TableCell,
  TableHeader,
  TableRow,
  TableFooter,
} from "@/components/ui/table";

import type { Items } from "@/types/order";

export default function OrderItemsTable({ items }: { items?: Items[] }) {

  return (
    <div className="overflow-hidden rounded-xl border border-gray-200 bg-white dark:border-white/[0.05] dark:bg-white/[0.03]">
      <div className="max-w-full overflow-x-auto">
        <Table>
          {/* Table Header */}
          <TableHeader className="border-b border-gray-100 dark:border-white/[0.05]">
            <TableRow>
              <TableCell
                isHeader
                className="px-5 py-3 font-medium text-gray-500 text-start text-theme-xs dark:text-gray-400"
              >
                Quantity
              </TableCell>
              <TableCell
                isHeader
                className="px-5 py-3 font-medium text-gray-500 text-start text-theme-xs dark:text-gray-400"
              >
                Description
              </TableCell>
              <TableCell
                isHeader
                className="px-5 py-3 font-medium text-gray-500 text-start text-theme-xs dark:text-gray-400"
              >
                unitPrice
              </TableCell>
            </TableRow>
          </TableHeader>

          <TableBody className="divide-y divide-gray-100 dark:divide-white/[0.05]">
            {Array.isArray(items) && items.map((item, idx) => (
              <TableRow key={idx}>
                <TableCell className="px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400">
                  {item.quantity}
                </TableCell>
                <TableCell className="px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400">
                  {item.description}
                </TableCell>
                <TableCell className="px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400">
                  {item.unitPrice}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
          <TableFooter>
            <TableRow>
              <TableCell colSpan={2} className="px-8 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400">Total</TableCell>
              <TableCell className="px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400">$
                {Array.isArray(items)
                  ? items
                    .reduce(
                      (acc, item) => acc + item.quantity * item.unitPrice,
                      0
                    )
                    .toFixed(2)
                  : "0.00"}</TableCell>
            </TableRow>
          </TableFooter>
        </Table>
      </div>
    </div>
  );
}
