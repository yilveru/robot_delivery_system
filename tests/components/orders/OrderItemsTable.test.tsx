import React from 'react'
import { render, screen } from '@testing-library/react'
import OrderItemsTable from '@/components/orders/OrderItemsTable'

describe('OrderItemsTable', () => {
  const mockItems = [
    { quantity: 2, description: 'Pizza', unitPrice: 10 },
    { quantity: 1, description: 'Soda', unitPrice: 5 },
  ]

  it('should render table headers', () => {
    render(<OrderItemsTable items={mockItems} />)

    expect(screen.getByText('Quantity')).toBeInTheDocument()
    expect(screen.getByText('Description')).toBeInTheDocument()
    expect(screen.getByText('unitPrice')).toBeInTheDocument()
  })

  it('should render table rows with items', () => {
    render(<OrderItemsTable items={mockItems} />)

    expect(screen.getByText('Pizza')).toBeInTheDocument()
    expect(screen.getByText('2')).toBeInTheDocument()
    expect(screen.getByText('10')).toBeInTheDocument()

    expect(screen.getByText('Soda')).toBeInTheDocument()
    expect(screen.getByText('1')).toBeInTheDocument()
    expect(screen.getByText('5')).toBeInTheDocument()
  })

  it('should calculate and render total', () => {
    render(<OrderItemsTable items={mockItems} />)

    const total = (2 * 10 + 1 * 5).toFixed(2)
    expect(screen.getByText(`$${total}`)).toBeInTheDocument()
  })

  it('should show $0.00 when no items are passed', () => {
    render(<OrderItemsTable items={[]} />)
    expect(screen.getByText('$0.00')).toBeInTheDocument()
  })
})

it('should match snapshot', () => {
  const { asFragment } = render(
    <OrderItemsTable
      items={[
        { quantity: 1, description: 'Burger', unitPrice: 8 },
        { quantity: 2, description: 'Fries', unitPrice: 3 },
      ]}
    />
  )

  expect(asFragment()).toMatchSnapshot()
})