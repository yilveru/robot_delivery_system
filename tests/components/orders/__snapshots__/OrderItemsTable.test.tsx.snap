// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden rounded-xl border border-gray-200 bg-white dark:border-white/[0.05] dark:bg-white/[0.03]"
  >
    <div
      class="max-w-full overflow-x-auto"
    >
      <table
        class="min-w-full  undefined"
      >
        <thead
          class="border-b border-gray-100 dark:border-white/[0.05]"
        >
          <tr>
            <th
              class=" px-5 py-3 font-medium text-gray-500 text-start text-theme-xs dark:text-gray-400"
            >
              Quantity
            </th>
            <th
              class=" px-5 py-3 font-medium text-gray-500 text-start text-theme-xs dark:text-gray-400"
            >
              Description
            </th>
            <th
              class=" px-5 py-3 font-medium text-gray-500 text-start text-theme-xs dark:text-gray-400"
            >
              unitPrice
            </th>
          </tr>
        </thead>
        <tbody
          class="divide-y divide-gray-100 dark:divide-white/[0.05]"
        >
          <tr>
            <td
              class=" px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
            >
              1
            </td>
            <td
              class=" px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
            >
              Burger
            </td>
            <td
              class=" px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
            >
              8
            </td>
          </tr>
          <tr>
            <td
              class=" px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
            >
              2
            </td>
            <td
              class=" px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
            >
              Fries
            </td>
            <td
              class=" px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
            >
              3
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              class=" px-8 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
              colspan="2"
            >
              Total
            </td>
            <td
              class=" px-4 py-3 text-gray-500 text-start text-theme-sm dark:text-gray-400"
            >
              $14.00
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
